examples/gno.land/r/demo/disperse/disperse.gno

```diff
@@ -0,0 +1,122 @@
+package disperse
+
+import (
+	"std"
+
+	tokens "gno.land/r/demo/grc20factory"
+)
+
+// Check if number of addresses and coins match
+func checkNumAddrValMatch(addresses []std.Address, coins std.Coins) {
+	if len(addresses) != len(coins) {
+		panic(ErrNumAddrValMismatch)
+	}
+}
+
+// Check if caller correctly sent positive amount of coins
+func checkCoinAmount(coins std.Coins) {
+	for _, coin := range coins {
+		if coin.Amount <= 0 {
+			panic(ErrNegativeCoinAmount)
+		}
+	}
+}
+
+// DisperseGnot parses receivers and amounts and sends out gnot
+func DisperseGnot(addresses []std.Address, coins std.Coins) {
+	coinSent := std.GetOrigSend() // get Coins sent with call
+	caller := std.GetOrigCaller() // get tx sender
+
+	checkCoinAmount(coins)
+	checkNumAddrValMatch(addresses, coins)
+
+	// Get address of Disperse realm
+	realmAddr := std.CurrentRealm().Addr()
+
+	// Get Banker
+	banker := std.GetBanker(std.BankerTypeOrigSend)
+
+	var totalAmount int64
+	for _, coin := range coins {
+		totalAmount += coin.Amount
+	}
+
+	// Check if total amount of coins sent is bigger or equal to total amount of coins to be sent
+	if totalAmount > coinSent.AmountOf("ugnot") {
+		panic(ErrWrongAmount)
+	}
+
+	// Check if the realm has zero ugnot
+	if banker.GetCoins(realmAddr).AmountOf("ugnot") < coinSent.AmountOf("ugnot") {
```

Comment:
Is this case possible? If it is, maybe the check should be higher in code, to stop execution right away, since it is a critical error?

Reply:
Did you have any updates to this convo?

Reply:
sorry, i didn't provide a commit link because i switched that logic up a bunch of times throughout several commits
I ended up changing the check and placing it at the beginning of the function like you said